unified_planning/model/walkers/type_checker.py

```diff
@@ -83,6 +83,46 @@ def walk_param_exp(
         assert len(args) == 0
         return expression.parameter().type
 
+    def walk_always(
+        self, expression: FNode, args: List["unified_planning.model.types.Type"]
+    ) -> "unified_planning.model.types.Type":
+        assert expression is not None
+        assert expression.is_always()
+        assert len(expression.args) == 1
+        return BOOL
```

Comment:
In all those `return BOOL`, i think you should check what is actually returned in the `args`. For construction this should be equivalent, but it's better to check that the `args` is actually a `BOOL`, because if it is something else, you should return `None`